File: src/main/java/Managers/Checker.java
package Managers;

import AbstractContracts.GraphicDotContainer;
import AbstractContracts.MathFiguresChecker;
import DataTransfer.GraphicDots;

public class Checker implements MathFiguresChecker {
    public boolean checkHit(GraphicDots graphicDotContainer){
        return ( checkCircle(graphicDotContainer) || checkSquare(graphicDotContainer) || checkTriangle(graphicDotContainer) );
    }


    @Override
    public boolean checkCircle(GraphicDots graphicDotContainer) {
        Double x = graphicDotContainer.getParam("x");
        Double y = graphicDotContainer.getParam("y");
        Double r = graphicDotContainer.getParam("r");
        return (((x*x+y*y)<=Math.pow((r),2) && (x<=0)) && (y>=0));
    }

    @Override
    public boolean checkTriangle(GraphicDots graphicDotContainer) {
        Double x = graphicDotContainer.getParam("x");
        Double y = graphicDotContainer.getParam("y");
        Double  r = graphicDotContainer.getParam("r");
        return (( (y > -x-r) && (y <= 0) && (x <= 0) ));
    }

    @Override
    public boolean checkSquare(GraphicDots graphicDotContainer) {
        Double x = graphicDotContainer.getParam("x");
        Double y = graphicDotContainer.getParam("y");
        Double r = graphicDotContainer.getParam("r");
        return (( 0<=x && x<=r/2) &&(0<=y && y<=r));
    }

}

File: src/main/kotlin/Managers/Checker.kt
package org.example.Managers

public class Checker {
    public fun checkHit(x: Int, y: Int, r: Int): Boolean{
        // Кружок
        if (((((0-x)*(0-x)+(0-y))<=r*r) )){
            return true
        }
        // Треугольник
        else if ((x >= -r/2) && (y >= -r/2) && (2 * x + 2 * y >= -r) && x <= 0 && y <= 0){
            return true;
        }
        // Прямоугольник
        else if(x >= 0 && x <= r/2 && y >= 0 && y <= r){
            return true
        }
        else{
            return false;
        }
    }


}